scipy/linalg/_decomp_svd.py

```diff
@@ -108,7 +108,26 @@ def svd(a, full_matrices=True, compute_uv=True, overwrite_a=False,
     a1 = _asarray_validated(a, check_finite=check_finite)
     if len(a1.shape) != 2:
         raise ValueError('expected matrix')
-    m, n = a1.shape
```

Comment:
This looks wrong. I have no idea why CI isn't unhappy, but I see:
```
scipy/linalg/_decomp_svd.py:141: in svd
    max_mn, min_mn = (m, n) if m > n else (n, m)
E   UnboundLocalError: cannot access local variable 'm' where it is not associated with a value
```

Reply:
Ah, it's because of a cross-merge conflict with gh-20349. Lucky I rebased before testing this locally:)